src/main/java/org/apache/datasketches/tuple/HashTables.java

```diff
@@ -80,13 +78,12 @@ void fromArrays(final long[] hashArr, final S[] summaryArr, final int count) {
     count_ = count;
     lgTableSize_ = getLgTableSize(count);
 
-    S mySummary = null;
     summaryTable_ = null;
     hashTable_ = new long[1 << lgTableSize_];
     for (int i = 0; i < count; i++) {
       final long hash = hashArr[i];
       final int index = hashInsertOnly(hashTable_, lgTableSize_, hash);
-      mySummary = summaryArr[i];
+      final S mySummary = summaryArr[i];
```

Comment:
I suppose in this case it is fine not to make copies. The sketch can take ownership of the given objects, right?

Reply:
In this case, it is fine.  

There are 2 ways to initialize the HashTables class: `fromSketch(...tuple.Sketch<S>)` and `fromSketch(...theta.Sketch)`.  In both cases the elements of the internal summaries arrays are copies of the summaries of the presented sketch, or copies of the provided summary in the case of theta.

The method you ask about, `fromArrays(...)`, is a private method (now private :) ) and only called by the 2 `getIntersectHashTables(next*Sketch,...)` methods.   These two methods rely on the `SummarySetOperations<S>` class to perform the configured set operation with the internal state of `HashTables` and the provided `next*Sketch`.   

Meanwhile, the Javadoc of the `SummarySetOperations<S>` interface has a clear warning for implementers:

```
Caution: Do not modify the input Summary objects. Also do not return them directly,
   unless they are immutable (most Summary objects are not). For mutable Summary objects, it is
   important to create a new Summary object with the correct contents to be returned. Do not
   return null summaries.
```
Assuming that this interface is implemented correctly by the user, there should not be any Summary objects belonging to input sketches that end up in the internals of these set operations.  This means that the extra `copy()`, line 150, I added in the last review should not be necessary.

A further question is whether we need to make sure no internal summary objects end up in result sketches.  I'm going to have to think more about this, but I think we may be OK, as the internals of these set operations are frequently cleared, ... nonetheless :)